### PR TITLE
Added regex pattern to color map

### PR DESCRIPTION
**Pull request by ggetz**
Added the option to specify a regex to the color map and added the option to the sandcastle example for color map.

### CODE REVIEW

File: Source/Scene/ColorMapExpression.js [hunk 1]
```diff
@@ -109,6 +127,15 @@ define([
     ColorMapExpression.prototype.evaluate = function(feature) {
         var value = feature.getProperty(this._propertyName);
 
+        if (defined(this._pattern)) {
+            var match = this._pattern.exec(value);
+            if (defined(match)) {
+                value = match[1];
```
Comment (pjcozzi) [inline Source/Scene/ColorMapExpression.js]: Make a new local variable instead of assigning to `value` again (it will also default to `undefined`).

**Comment by pjcozzi** [comment]
Just that comment, this looks good, @ggetz.  Please check items off the roadmap, https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/2, as you finish them.

**Comment by ggetz** [comment]
Fixed and marked off on the roadmap.

**Comment by pjcozzi** [comment]
Looks good!

**Pull request merged**